gprofiler/profilers/perf.py

```diff
@@ -121,6 +132,7 @@ def _collapse_stack(comm: str, stack: str, insert_dso_name: bool = False) -> str
         m = FRAME_REGEX.match(line)
         assert m is not None, f"bad line: {line}"
         sym, dso = m.groups()
+        dso = dso.strip("[]")
```

Comment:
I haven't fully read the new regex pattern yet, but trying to understand here - are you expecting `[]` to be optionally around the dso? To be certainly around the dso?

I prefer the code to be explicit. If you always expect - then assert they are there, and remove them. If they are optional (but one set of `[]`) then remove at most one (`strip` might remove multiple, ...)

Reply:
My observation is that style of dso name in frames is much more well-defined.
It's either `([vdso])` or `(library_path)` or `(library_path (deleted))`. 3 well defined patterns we can build upon.

But that's good idea to make this pattern more explicit. I will also check if it still doesn't affect performance negatively.

Reply:
Checked performance. New regex is 15% faster on a large perf script (1.29us vs 1.51us per frame on average).